### ISSUE TITLE
Scenarios - Composition change

### ISSUE THREAD

**Issue by pautva** [micronutrientsupport/micronutrient-support-tool#506]
# Figma prototype status

![status-develop](https://user-images.githubusercontent.com/72566636/120622435-b5b98d00-c456-11eb-9b78-242652fb362d.png)

# **User story**

As a policymaker, I want to explore how food composition change affects the nutrition in a selected country.

## Location
<details>
<summary> Wireframe for the whole scenarios page</summary>
<br>

![Scenarios](https://user-images.githubusercontent.com/72566636/116717209-8edbe700-a9d0-11eb-8762-eaba8d5cea74.png)
</details>

### Scenario options - Composition change

Adjust the composition of a selected food item(s):

![image](https://user-images.githubusercontent.com/72566636/116723224-30663700-a9d7-11eb-8dac-66f11c6d5df8.png)

These options will affect the #503, #504 and #505.

<details >
<summary>No food item selected</summary>
<br>

![image](https://user-images.githubusercontent.com/72566636/117477393-de349100-af55-11eb-8a0d-537f3888f791.png)

</details>
<details >
<summary>Multiple food items added</summary>
<br>

![image](https://user-images.githubusercontent.com/72566636/116730738-c3579f00-a9e0-11eb-9290-ba6b618cd0a8.png)

</details>

# Location

Diet Data > Explore Simple dietary change scenarios

**Part of scenarios page:**
#499 

**Part of component:**
#501

**Related:**
#507 Consumption change
#508 Food item comparison

# Reference

<details>
<summary>Other Scenarios issues</summary>
<br>

![gloomap_002be100](https://user-images.githubusercontent.com/72566636/120618179-ba7c4200-c452-11eb-86d8-919e5051f0a7.png)

</details>

# Acceptance criteria

<details open>
<summary> Acceptance criteria</summary>
<br>

- [x] Select Food item group
- [x] Select a food item from a food item group
- [ ] Option to select All items from a food item group
- [x] Display current composition of the food item in mg/kg
- [x] Specify scenario composition by changing value manually
- [x] Increase or decrease the scenario  value by 5%
- [ ]  While no food item is selected, display a message on the map and table to users that scenario details need to be selected first
- [x] (NICE TO HAVE) Add multiple food items to a scenario
- [x] (NICE TO HAVE) Remove extra items

</details>

**Comment by jon571** [comment]
If you do select the "All Items" Option for a food group what appears on the right?

I don't see how you could show the current or scenario composition.   Maybe it will have to be "Scenario Composition Change", with a % value displayed, indicating that all of them are changed by the displayed percentage.

I assume that same item can't be selected twice also, so once selected it will need removing from the available options.  Consequently, if "All items" is selected then none of the individual items can be added individually.

**Comment by pautva** [comment]
If we select "All Groups" or "All Food Items", we might want to display the average value for the selection if possible. Then, could we change the average scenario composition/composition for that group?

Yes, we want to select a food item or group only once.

**Comment by Fuhji** [comment]
Blocked until #501 scaffolded options are available

**Comment by jon571** [comment]
> If we select "All Groups" or "All Food Items", we might want to display the average value for the selection if possible. Then, could we change the average scenario composition/composition for that group?
> 
> Yes, we want to select a food item or group only once.

I'd like to see @bgsandan 's feedback on this.  It would mean that the "current value" endpoints for composition and consumption would need to take an array of food item ids too.

**Comment by bgsandan** [comment]
Interesting point.  I think for the "All" options then yes we will need to display something different (e.g. an average/sum) for the 'current' value.  Are we trying to shoe-horn too much into this endpoint though?  Returning an aggregation in some cases and a single value in others seems like poor separation of concerns

Does it muddy things too much to go back to the previous conversation around allowing the front-end to call EITHER a separate aggregate endpoint that accepts a food_group id and provides an aggregate value (of some form) OR the existing endpoint for individual fooditems?
